sdk/storagecache/Microsoft.Azure.Management.StorageCache/src/Generated/Models/Cache.cs

```diff
@@ -140,11 +158,46 @@ public Cache(object tags = default(object), string id = default(string), string
         [JsonProperty(PropertyName = "properties.upgradeStatus")]
         public CacheUpgradeStatus UpgradeStatus { get; set; }
 
+        /// <summary>
+        /// Gets or sets specifies network settings of the cache.
+        /// </summary>
+        [JsonProperty(PropertyName = "properties.networkSettings")]
+        public CacheNetworkSettings NetworkSettings { get; set; }
+
+        /// <summary>
+        /// Gets or sets specifies encryption settings of the cache.
+        /// </summary>
+        [JsonProperty(PropertyName = "properties.encryptionSettings")]
+        public CacheEncryptionSettings EncryptionSettings { get; set; }
+
+        /// <summary>
+        /// Gets or sets specifies security settings of the cache.
+        /// </summary>
+        [JsonProperty(PropertyName = "properties.securitySettings")]
+        public CacheSecuritySettings SecuritySettings { get; set; }
+
         /// <summary>
         /// Gets or sets SKU for the Cache.
         /// </summary>
         [JsonProperty(PropertyName = "sku")]
         public CacheSku Sku { get; set; }
 
+        /// <summary>
+        /// Validate the object.
+        /// </summary>
+        /// <exception cref="ValidationException">
+        /// Thrown if validation fails
+        /// </exception>
+        public virtual void Validate()
+        {
+            if (NetworkSettings != null)
+            {
+                NetworkSettings.Validate();
+            }
+            if (EncryptionSettings != null)
+            {
+                EncryptionSettings.Validate();
+            }
+        }
```

Comment:
From 2020-03-01 moving forward, networkSettings should always be a cache property. I think if it's null, there's a problem.

Should we have a similar SecuritySettings.Validate() call? Is there any other validation that isn't done elsewhere that can/should be done here? Example: entering 12289 for cache size instead of a supported value, etc.

This may be a no-op, especially since this generated code. If a change is needed, it might be in an upcoming API definition.

Reply:
hrmm yes we will have to check what makes auto rest generate these depending upon swagger.